### DISCUSSION TITLE
[V3] - How do I test lazy loaded components?

### DISCUSSION THREAD

**Discussion by jesperbeisner** [livewire/livewire#6719] [1. Help] [answered]
Hey all 👋,

I wanna test lazy loaded components but the only stuff that is rendered in my tests is the placeholder. So how do I "dispatch" some "load/on-load" state so I can assert on the real rendered html?

Thanks 🙃

**Comment by pikseli** [comment, marked as answer]
Only workaround I found was to disable lazy loading on tests by setting the `lazy` property to `false`.

```php
Livewire::test(ShowComments::class, ['lazy' => false])
    ->assertSee('Some actual content.');
```

**Comment by PhiloNL** [comment]
Please take a look at the Livewire test :) This contains examples of how to assert the rendered HTML.
https://github.com/livewire/livewire/blob/main/src/Features/SupportLazyLoading/BrowserTest.php

**Comment by danjdewhurst** [comment]
What about when doing a page request and wanting to test for the presence of renders content from a lazy loaded component? As in not using `Livewire::test/visit` but a `$this->get()` etc.

Edit: Just found the `Livewire::withoutLazyLoading` - this is nice but it would be cool to be able to turn it off for an entire test and/or request.

**Reply by megawubs** [reply]
This should be the accepted answer
